src/document/DocumentAnnotator.ts

```diff
@@ -0,0 +1,82 @@
+// @flow
+import BaseAnnotator, { Options } from '../common/BaseAnnotator';
+import BaseManager from '../common/BaseManager';
+import RegionManager from '../region/RegionManager';
+import { ANNOTATOR_EVENT, CLASS_ANNOTATIONS_LOADED } from '../constants';
+
+export default class DocumentAnnotator extends BaseAnnotator {
+    annotatedEl?: HTMLElement;
+
+    annotations: Record<string, []> = {};
+
+    managers: Map<string, BaseManager[]> = new Map();
+
+    constructor(options: Options) {
+        super(options);
+
+        this.hydrate(options);
+    }
+
+    getPages(): HTMLElement[] {
+        // TODO: Inject page/container elements from Preview SDK rather than DOM?
+        return this.annotatedEl ? Array.from(this.annotatedEl.querySelectorAll('.page')) : [];
+    }
+
+    hydrate({ file }: Options): void {
+        this.api
+            .fetchAnnotations(file.file_version.id)
+            .then(() => {
+                // TODO: Normalize response, set in store, and render
+            })
+            .catch(error => {
+                this.emit(ANNOTATOR_EVENT.error, error);
+            });
+    }
+
+    init(scale: number): void {
+        super.init(scale);
+
+        if (!this.rootEl) {
+            return;
+        }
+
+        this.annotatedEl = this.rootEl.querySelector('.bp-doc') as HTMLElement;
+        this.annotatedEl.classList.add(CLASS_ANNOTATIONS_LOADED);
+
+        this.render();
+    }
+
+    render(): void {
+        this.getPages()
+            .filter(({ dataset }) => dataset.loaded && dataset.pageNumber)
+            .forEach(pageEl => this.renderPage(pageEl));
+    }
+
+    renderPage(pageEl: HTMLElement): void {
+        const { pageNumber = '1' } = pageEl.dataset;
+        const annotations = this.annotations[pageNumber];
+        const managers = this.managers.get(pageNumber) || [];
+
+        // Destroy any managers that were attached to page elements that no longer exist
+        if (managers.some(manager => !manager.exists(pageEl))) {
```

Comment:
Maybe it's more clear once the other code you've written comes in, but just looking at this, the management of managers (🤣) seems excessive. Is there a tradeoff to have one typed Manager that can render in any page rather than one typed Manager per page?

Reply:
Let's chat about it offline.